source/common/network/apple_dns_impl.cc

```diff
@@ -87,22 +87,55 @@ ActiveDnsQuery* AppleDnsResolverImpl::resolve(const std::string& dns_name,
                                               DnsLookupFamily dns_lookup_family,
                                               ResolveCb callback) {
   ENVOY_LOG(debug, "DNS resolver resolve={}", dns_name);
-  std::unique_ptr<PendingResolution> pending_resolution(
-      new PendingResolution(*this, callback, dispatcher_, main_sd_ref_, dns_name));
 
-  DNSServiceErrorType error = pending_resolution->dnsServiceGetAddrInfo(dns_lookup_family);
-  if (error != kDNSServiceErr_NoError) {
-    ENVOY_LOG(warn, "DNS resolver error ({}) in dnsServiceGetAddrInfo for {}", error, dns_name);
-    return nullptr;
+  Address::InstanceConstSharedPtr address{};
+  try {
+    // After production testing with this resolver implementation it became empirically demonstrable
```

Comment:
Is this actually an issue in practice? When are you resolving IP addresses and not names? Trying to understand if it matters if we pass IP addresses to resolvers or not?

/wait-any

Reply:
Yep, this is the line of questioning @goaway @rebello95 and myself discussed over slack yesterday. 

At Lyft, and I am guessing the vast majority of envoy users, we do not give already resolved addresses to a dns based cluster or to the dynamic forward proxy system. However, if it were to happen, this resolver might exhibit behavior that is different from the c-ares resolver. 

I think one of the most interesting cases (that @goaway brought up) was receiving a redirect with a resolved IP. I need to verify if the url parser envoy uses in that case allows an IP.

Reply:
OK this all seems pretty theoretical to me. I would personally just drop this PR until there is a known issue.

Reply:
Yeah, certainly more theoretical than practical. The desire was to be defensive and provide parity of behavior between this and the c-ares resolver.

Reply:
OK then I would recommend closing this for now. I don't think it's worth it.

Reply:
I do think we should merge this. The strongest reason for that is simply that it's what the c-ares resolver does, it's what most other resolver interfaces do, and by the principle of the least surprise, it seems like it's what this should do. But also I think it's worth noting:

1) We did encounter it - in our case arguably because of invalid configuration, but it's not entirely uncommon for people to use IPs interchangeably with domain names. It took some time to debug and caused notable degradation (a 30s delay to all networking).
2) Using an IP address in a URL *is* valid, and it will hit this as well (tested/confirmed).
3) Sending DNS queries for IPs is at best ill-formed and is maybe(?) a security flaw.

Just my take.

Reply:
I'm fine if you want to merge this "nice to have," please fix the overall testing of this code and then we can revisit (or just merge the test improvements into this PR). I believe @junr03 has a plan for this.

/wait

Reply:
Sure! fwiw the additional code here is properly tested _because_ it doesn't call Apple's API 😂 . But yeah, I will push the testing we discussed in #13641 asap.

Reply:
test suite PR here: https://github.com/envoyproxy/envoy/pull/13780. Will land that and then update this one.